packages/core/lib/debug/interpreter.js

```diff
@@ -56,6 +56,9 @@ class DebugInterpreter {
     const currentSourceId = currentLocation.source
       ? currentLocation.source.id
       : null;
+    const currentCompilationId = currentLocation.source
```

Comment:
Is there any way to avoid compilation being a necessary part of the definition of a breakpoint?

Reply:
Well, without a compilation, a source index no longer uniquely specifies the source, so I think we need it, yes.  Of course we could use source *ID*s, but:

1. (The stupid reason) Introducing source IDs into the debugger would require renaming things so it doesn't call source indexes source IDs :P
2. (The better reason) We won't have source IDs if this is a shimmed compilation.

I mean, if we know that real compilations, rather than shimmed ones, will always have source IDs, then we could use a weird hybrid solution where we use source IDs when we have them, but if we don't we use source indices, and we omit the compilation because there should only be one (shimmed) compilation in that case?  That sounds even ickier though.

But honestly I don't really see anything wrong with including compilation?  Anyone coding against the debugger library can always just do what I did and, y'know, get the current compilation ID from the appropriate selector.  Yes, that's is a breaking change, but given all the selector changes this is a breaking PR anyway.  And doing it the other way -- based on source IDs rather than indexes, were we to add those -- would be breaking too.

Reply:
> then we could use a weird hybrid solution where we use source IDs when we have them, but if we don't we use source indices

Could we just use source indices as source IDs in that situation?

Reply:
Yes, I guess we could do that.  Of course then I'm going to have to go through and make source ID and source index separate concepts in the debugger. :P

Honestly I think this is OK as it is, but I guess it would be nicer to have just a single source ID instead of a pair (compilation, index).

Reply:
Oh, here's a thought: What if we put source IDs in the debugger, but we gave them a different name, like `uniqueId` or something?  (To distinguish from the ordinary, non-unique IDs, i.e., indices...)  Yes, that's terrible, but then I wouldn't have to sort through everywhere a source index is used and change it...

(This would still probably be breaking though.  I think the only way to keep `compilation` out in a fully non-breaking way would be to do things the hard way. :-/ )

Reply:
OK, talking things over with @gnidan, I'm leaving this as-is for now...